src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -2196,6 +2224,25 @@ export class SavedObjectsRepository {
       );
     }
   }
+
+  /** Validate a migrated doc against the registered saved object type's schema. */
+  private validateObjectAttributes<T>(type: string, doc: SavedObjectSanitizedDoc<T>) {
+    const savedObjectType = this._registry.getType(type);
+    if (!savedObjectType?.schemas) {
+      return;
+    }
+
+    const validator = new SavedObjectsTypeValidator({
+      type,
+      validationMap: savedObjectType.schemas,
+    });
```

Comment:
NIT/optional: do we really need to instantiate this on every call to `validateObjectAttributes`? can't we instantiate a validator per type when the repository is created?